drivers/hmis/app/graphql/mutations/delete_client.rb

```diff
@@ -0,0 +1,74 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class DeleteClient < BaseMutation
+    argument :id, ID, required: true
+    argument :confirmed, Boolean, 'Whether warnings have been confirmed', required: false
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(id:, confirmed: false)
+      client = Hmis::Hud::Client.find_by(id: id)
+
+      warnings, resolvable_enrollments = check_enrollments(client, ignore_warnings: confirmed)
+
+      return { client: nil, errors: warnings } if warnings.present?
+
+      default_delete_record(
+        record: client,
+        field_name: :client,
+        permissions: :can_delete_clients,
+        after_delete: -> do
+          resolvable_enrollments.each do |enrollment|
+            enrollment.household_members.where.not(personal_id: client.personal_id).first&.update!(relationship_to_ho_h: 1)
```

Comment:
Will each call to `household_members` here hit the db again, even though we query for that exact thing on line 41 already?

Reply:
Yes it would since `household_members` is just a method that will then fire a query. Since `household_members` are just enrollments themselves, I just passed the household member enrollments as resolvable enrollments, which removed the need to call `household_members` again.

Best case I would probably prefer to see `household_members` as a `has_many` association so we can preload it, but it looks like whatever version of rails we're using doesn't support preloading instance-specific association scopes, so I don't think we'd get that benefit right now anyway, so I left it as a method.

Reply:
Looks good, thanks for updating that